### ISSUE TITLE
"sh: tput: not found" and "Warning: A non-numeric value encountered"

### ISSUE THREAD

**Issue by EvgenyOrekhov** [wapmorgan/PhpCodeFixer#21]
I use PhpCodeFixer inside a Docker container (I use the official `php:fpm-alpine` Docker image), and when I run `vendor/bin/phpcf modules` I get the following output:

```text
Max file size set to: 1.000 MiB
Scanning modules ...
sh: tput: not found

Warning: A non-numeric value encountered in /app/vendor/wapmorgan/php-code-fixer/bin/phpcf on line 74
 PHP |             Type |                      File:Line | Issue
Peak memory usage: 1.758 MB
```

**Comment by EvgenyOrekhov** [comment]
Looks like the `stty` utility is more widespread than `tput`. It is available in Alpine Linux.

You can get terminal size like this:

```text
stty size
49 127
```

The first value is rows, the second is columns.

**Comment by EvgenyOrekhov** [comment]
@wapmorgan Thanks, but I can't check it out due to https://github.com/wapmorgan/PhpCodeFixer/issues/24.

**Comment by EvgenyOrekhov** [comment]
@wapmorgan dev-master works fine!

**Comment by wapmorgan** [comment]
Fixed in last release.

**Comment by EvgenyOrekhov** [comment]
@wapmorgan I'm getting `stty: standard input` message now.

Example:

```
$ ./vendor/bin/phpcf src/
Scanning /home/user/project/src ...
stty: standard input

Folder /home/user/project/src

Analyzer has not detected any problems in your code.
Peak memory usage: 2.157 MB
```

Maybe it's related to the fact that I run PHP using Docker, not sure.

It doesn't really bother me though.

**Comment by MetalArend** [comment]
@wapmorgan Thanks for fixing this, it makes it easier to install on an alpine machine. Glad to see this is already resolved!

@EvgenyOrekhov Are you adding the `--tty` flag to your `docker run` command?
But this might need resolving inside the package, as things should probably fall back to some defaults instead of going wrong when a tty is not available - one example might be running this tool in some CI environment for example.

**Comment by EvgenyOrekhov** [comment]
@MetalArend I run `docker-compose exec php ./vendor/bin/phpcf src/`, it allocates a TTY by default.

**Comment by MetalArend** [comment]
You're right, `docker-compose exec` does allocate a TTY by default. What happens if you add the `-T` flag?

**Comment by EvgenyOrekhov** [comment]
@MetalArend The `stty: standard input` message goes away. Thanks!

**Comment by wapmorgan** [comment]
Is there anything I can do to fix problem within phpcf?

**Comment by EvgenyOrekhov** [comment]
@wapmorgan Perhaps it is possible to detect whether a TTY is allocated and don't execute the `stty` command if it is. I don't have enough expertise in this matter, but maybe something like [posix_isatty()](http://php.net/manual/en/function.posix-isatty.php) could be of use here.

**Comment by michabbb** [comment]
https://github.com/dotnet/cli/issues/8311#issuecomment-354870954
